components/common/AsyncBoundary.tsx

```diff
@@ -0,0 +1,16 @@
+import React, { Suspense } from 'react';
+import { ErrorBoundary } from 'react-error-boundary';
+
+interface AsyncBoundaryProps {
+  errorFallback?: React.ReactElement<unknown, string | typeof React.Component | React.FunctionComponent> | null;
+  loadingFallback?: React.ReactNode;
```

Comment:
💊 
`errorFallback` 타입은 `React.ReactNode`로 줄일 수 있을 것 같습니다!

그리고 외부에서 주입받을 컴포넌트의 타입을 정의할 때 많이 사용되는데 [요 블로그](https://www.howdy-mj.me/react/react-node-and-jsx-element)를 참고하시면 더욱 좋은 타입을 선언할 수 있을 것 같습니다!

![image](https://github.com/codeit-bootcamp-frontend/4-Weekly-Mission/assets/53929065/de4675e8-c75c-43cb-ba13-120566352cbe)


아래는 저라면 이렇게 타입을 줄 것 같아서 서제스쳔 드립니다!

```suggestion
  errorFallback?: ComponentProps<typeof ErrorBoundary>['fallback']
  loadingFallback?: ComponentProps<typeof Suspense>['fallback']
```

`ComponentProps<typeof ErrorBoundary>['fallback']`과 `ComponentProps<typeof Suspense>['fallback']`을 사용함으로써, 각각의 컴포넌트에 대한 정확한 타입을 상속받습니다.

`React.ReactNoe`로 할 수 있었지만 위처럼 작성한 이유는 `ErrorBoundary`와 `Suspense` 컴포넌트의 타입이 업데이트될 경우, 이러한 변경사항이 자동으로 AsyncBoundary 컴포넌트의 타입에도 반영되길 바라는 마음에서 작성했습니다 (유지보수성)